### PR TITLE
py-fenics-ufl: update version and add test

### PR DESCRIPTION
**Pull request by garth-wells**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-fenics-ufl/package.py [hunk 1]
```diff
@@ -36,7 +37,17 @@ class PyFenicsUfl(PythonPackage):
     )
     version("2016.2.0", tag="ufl-2016.2.0", commit="962d56f65821fb9c50ca4a5a858882c472243431")
 
-    depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-setuptools@58:", type=("build", "run"), when="@2022.1.0:")
-    depends_on("py-setuptools@40:", type=("build", "run"))
+    depends_on("python@3.7:", when="@2016.2.0:2023.1.1.post0", type=("build", "run"))
+    depends_on("python@3.8:", when="@2023.2.0:", type=("build", "run"))
+
+    depends_on("py-setuptools@58:", when="@2022.1.0:", type=("build", "run") )
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-ufl/package.py]: setuptools 62+ now, also only needed at build time

File: var/spack/repos/builtin/packages/py-fenics-ufl/package.py [hunk 2]
```diff
@@ -36,7 +37,17 @@ class PyFenicsUfl(PythonPackage):
     )
     version("2016.2.0", tag="ufl-2016.2.0", commit="962d56f65821fb9c50ca4a5a858882c472243431")
 
-    depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-setuptools@58:", type=("build", "run"), when="@2022.1.0:")
-    depends_on("py-setuptools@40:", type=("build", "run"))
+    depends_on("python@3.7:", when="@2016.2.0:2023.1.1.post0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-ufl/package.py]: You can remove this line, Spack only supports Python 3.7+ already

File: var/spack/repos/builtin/packages/py-fenics-ufl/package.py [hunk 3]
```diff
@@ -36,7 +37,17 @@ class PyFenicsUfl(PythonPackage):
...
+    depends_on("python@3.8:", when="@2023.2.0:", type=("build", "run"))
+
+    depends_on("py-setuptools@58:", when="@2022.1.0:", type=("build", "run") )
+    depends_on("py-setuptools@40:", when="@2016.2.0:2021.1.0", type=("build", "run"))
     depends_on("py-numpy@1.21:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-ufl/package.py]: Where does 1.21+ come from?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by garth-wells** [comment]
@adamjstewart Thanks for the comments, I think this PR is ready now.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**